float64  current_x0
float64  current_x1
